tests/page/frame-hierarchy.spec.ts

```diff
@@ -154,7 +156,10 @@ it('should report frame from-inside shadow DOM', async ({ page, server }) => {
   expect(page.frames()[1].url()).toBe(server.EMPTY_PAGE);
 });
 
-it('should report frame.name()', async ({ page, server }) => {
+it('should report frame.name()', async ({ page, server, browserName, channel }) => {
+  it.skip(browserName === 'firefox' && channel?.startsWith('moz-firefox'), 'frame.name() is racy with BiDi');
+  it.skip(channel?.startsWith('bidi-chrom'), 'frame.name() is racy with BiDi');
```

Comment:
FWIW, this test could be fixed by using
```
expect((await (await page.frames()[1].frameElement()).getAttribute('id'))).toBe('theFrameId');
expect((await (await page.frames()[2].frameElement()).getAttribute('name'))).toBe('theFrameName');
```

Reply:
As the test's title suggests it validates `frame.name()` behavior, so the code should invoke `name()` at some point.